### ISSUE TITLE
Question about `tol`

### ISSUE THREAD

**Issue by suiyoubi** [dilevin/CSC417-a3-finite-elements-3d#15]
Are we supposed to use the same `tol` for checking for convergence and checking for if alpha is too small?
Also what is the recommended value for `tol`?

**Comment by honglin-c** [comment]
Just a bit confused, what's the `tol` here?  Are you referring to the stopping condition of line search?

**Comment by suiyoubi** [comment]
Yeah I am referring to 1) the stopping condition of line search AND 2) the stopping condition for iteration (which is the Step2 "Check for convergence"of gradient based optimization method in the video). Should I use the same `tol` value?

**Comment by honglin-c** [comment]
1) For the stopping condition of the backtracking line search (i.e., alpha), you can use the stopping condition in the [Wikipedia](https://en.wikipedia.org/wiki/Backtracking_line_search) page with the following parameter values: alpha (initial step length) = 1, p (scaling factor) = 0.5, c (ensure sufficient decrease) = 1e-8.
2) For the convergence of newton's method, you can use a maximum of five (5) iterations (and stop when the norm of the gradient is less than, for example, 1e-1). 
(These parameters can be found at the end of the readme. :) )

**Comment by suiyoubi** [comment]
Yeah, the `tol` variable I was referring is 1e-1 in your context, and I think it is not mentioned in the readme.
In the video, it also describes another stopping condition for backtracking line search, which is when the `alpha` has been decresed significantly small enough (by compared to a variable also named `tol` in the video timestamped at 20:32). My question is should I also use 1e-1 for this comparasion as well ?

**Comment by honglin-c** [comment]
Yeah you are right. Neither of these is mentioned in the readme. Thanks for the reminder! 
For alpha, I think using `alpha > 1e-8` as a stopping condition should work. Or actually, just use any value that is "significantly" small enough. 1e-1 is a bit too large.  (Trying to find a suggested value on the Wikipedia page but failed. :( )

**Comment by suiyoubi** [comment]
Got it. Thanks!
